src/Microsoft.TemplateEngine.Cli/GitSource.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Cli
+{
+    internal class GitSource
+    {
+        public string GitUrl { get; set; }
+        public string SubFolder { get; set; }
+        public string RepositoryName { get; set; }
+
+        public GitSource(string gitUrl, string subFolder, string repoName)
+        {
+            GitUrl = gitUrl;
+            SubFolder = subFolder;
+            RepositoryName = repoName;
+        }
+
+        public static bool TryParseGitSource(string spec, out GitSource package)
+        {
+            package = null;
+            int gitIndex = -1;
+
+            if (string.IsNullOrEmpty(spec) || (gitIndex = spec.IndexOf(".git", StringComparison.OrdinalIgnoreCase)) < 0)
+            {
+                return false;
+            }
+            else
+            {
+                int index = gitIndex + 4;
```

Comment:
In cases like these, I prefer declaring a local const for the string (in this case ".git") and accessing the length instead of having the value of the length appearing as a magic number